### PR TITLE
feat(JobCard): add more track link type and refactoring

### PR DESCRIPTION
**Pull request by cm-seekasia**
Related to #311
Add company, jobTitle and location to `trackLinkClicked`.
Refactoring

## How to test
Go to JobCard, checked `Custom Link` and `Job Title Component`
1. Click company name and check console.log
2. Click job title and check console.log
3. Click location and check console.log

### CODE REVIEW

File: react/JobCard/components/CompanyLink/CompanyLink.js [hunk 1]
```diff
@@ -3,14 +3,15 @@ import PropTypes from 'prop-types';
 import styles from './CompanyLink.less';
 import { getParts } from '../../jobCardHelper.js';
 import defaultLink from '../Link/Link';
+import { trackLinkType } from '../../JobCard';
```
Comment (mengtzu) [inline react/JobCard/components/CompanyLink/CompanyLink.js]: Are you using this here?

**Comment by mengtzu** [comment]
🤔 Not sure I like explicit tracking in a design system - typically the way this has been done in the past is with link renderer props.  Otherwise you could make it a more generic (and optional) `onClick` sort of thing.

File: react/JobCard/components/CompanyLink/CompanyLink.js (hunk 1 above)
Comment (cm-seekasia) [reply react/JobCard/components/CompanyLink/CompanyLink.js]: Yes, in line 37

**Comment by cm-seekasia** [comment]
> 🤔 Not sure I like explicit tracking in a design system - typically the way this has been done in the past is with link renderer props. Otherwise you could make it a more generic (and optional) `onClick` sort of thing.

The original approach is to add a props(linkType) to `LinkComponent` #310. But @kosanna suggested that adding `trackLinkClicked` for doing that instead #311. Yes, I agree that `trackLinkClicked` can be more generic, like `onClick`. Those 3 approaches are ok for me, each of them have their own good. 
@mengtzu @kosanna Any comment?

**Comment by kosanna** [comment]
I do not want onClick clashing everywhere, it will end up being injected into onClick regardless, however I'd rather give it a separate name, because it's confusing if prop onClick is coming in as a prop on JobCard

**Comment by kosanna** [comment]
Yeah we could call it anyhow, but just make sure it's renamed everywhere consistently. onClick is confusing because if you pass it to <JobCard onClick={...} /> it intuitively looks like you're trying to do something when people just click on the whole jobcard

**Pull request merged**